editor/scene_tree_dock.cpp

```diff
@@ -2224,17 +2224,111 @@ void SceneTreeDock::_toggle_placeholder_from_selection() {
 	}
 }
 
+void SceneTreeDock::_update_all_gizmos(Node *p_node) {
+	Node3DEditor::get_singleton()->update_all_gizmos(p_node);
+}
+
+void SceneTreeDock::_reparent_nodes_to_root(Node *p_root, Array p_nodes, Node *p_owner) {
+	List<Node *> nodes;
+	for (int i = 0; i < p_nodes.size(); i++) {
+		Node *node = Object::cast_to<Node>(p_nodes[i]);
+		ERR_FAIL_COND(node == nullptr);
+		nodes.push_back(node);
+	}
+
+	for (Node *node : nodes) {
+		node->set_owner(p_owner);
+		if (node != p_root && node->get_owner() == p_owner && node->get_parent()->get_owner() != p_owner) {
+			List<Node *> owned;
+			node->get_owned_by(p_owner, &owned, true);
+			node->reparent(p_root);
+
+			for (Node *F : owned) {
+				F->set_owner(p_owner);
+			}
+		}
+	}
+}
+
+void SceneTreeDock::_reparent_nodes_to_paths_with_transform_and_name(Node *p_root, Array p_nodes, Array p_paths, Array p_transforms, Array p_names, Node *p_owner) {
+	ERR_FAIL_COND(p_nodes.size() != p_paths.size());
+	ERR_FAIL_COND(p_nodes.size() != p_transforms.size());
+	ERR_FAIL_COND(p_nodes.size() != p_names.size());
+
+	for (int i = 0; i < p_nodes.size(); i++) {
+		Node *node = Object::cast_to<Node>(p_nodes[i]);
+		ERR_FAIL_COND(node == nullptr);
+		NodePath np = p_paths[i];
+		Node *parent_node = p_root->get_node_or_null(np);
+		ERR_FAIL_COND(!parent_node);
+
+		List<Node *> owned;
+		node->get_owned_by(p_owner, &owned, true);
+		node->reparent(parent_node);
+		node->set_name(p_names[i]);
+
+		for (Node *F : owned) {
+			F->set_owner(p_owner);
+		}
+	}
+}
+
 void SceneTreeDock::_toggle_editable_children(Node *p_node) {
 	if (p_node) {
```

Comment:
Since you are changing most of this function anyway, you can replace this block with an early return instead.
```suggestion
	if (!p_node) {
		return;
	}
```